odata-to-abstract-sql.ometajs

```diff
@@ -755,30 +791,23 @@ var generateShortAliases = function(clientModel) {
 	var traverseNodes = function(str, node) {
 		_.each(node, function(value, key) {
 			if (key === '$suffix') {
-				var lowerCaseResourceName = str + value;
-				var origResourceName = _.find(resourceNames, function(resourceName) {
-					return resourceName.toLowerCase() === lowerCaseResourceName;
+				var lowerCaseAliasPart = str + value;
+				var origAliasPart = _.find(aliasParts, function(aliasPart) {
+					return aliasPart.toLowerCase() === lowerCaseAliasPart;
 				});
-				shortAliases[origResourceName] = origResourceName.slice(0, str.length);
+				var length = str.length;
+				if (str[str.length - 1] === ' ') {
+					// Add an extra character so we don't end on a space
+					length += 1;
```

Comment:
It's for a case like
```
person candied fruit
pilot can fly plane
```
you have the verbs `candied` and `can fly`, so the shortest unique one for each is `cand` and `can `, now whilst neither is particularly pretty regardless it's at least slightly nicer to use `can f` instead of `can `, so we increase the length of the string we take by 1 in order to include a letter (since spaces are collapsed to one)